### PR TITLE
[Merged by Bors] - Unify common dependencies

### PR DESCRIPTION
**Pull request by c410-f3r**
Continuation of #3408

To the best of my knowledge, all common dependencies are now sharing the same workspace definition.

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -1,4 +1,5 @@
 [workspace]
```
Comment (c410-f3r) [inline Cargo.toml:1]: Besides the inclusion of more workspace dependencies, some items were re-ordered to make manual looks-up a bit easier

**Comment by c410-f3r** [comment]
Looks like https://github.com/infinyon/fluvio/actions/runs/5754781767/job/15601307131 was spurious but now https://github.com/infinyon/fluvio/actions/runs/5788343499/job/15687610521?pr=3442 is showing a version mismatch. Needs more investigation on my side.

**Comment by c410-f3r** [comment]
@EstebanBorai CI looks OK now

**Review by EstebanBorai — APPROVED** [review]
Thanks @c410-f3r! Nice work!

File: crates/fluvio-test-case-derive/Cargo.toml [hunk 2]
```diff
@@ -17,4 +17,4 @@ quote = { workspace = true }
 syn = { workspace = true, features = ["full"]}
 
 [dev-dependencies]
-trybuild = {git = "https://github.com/infinyon/trybuild", branch = "check_option"}
+trybuild = { workspace = true }
```
Comment (sehz) [inline crates/fluvio-test-case-derive/Cargo.toml:20]: this is not same `trybuild`  keep as is

**Review by sehz — CHANGES_REQUESTED** [review]
all looks good except `trybuild` on a repo

**Review by sehz — APPROVED** [review]
LGTM.  Thanks

**Comment by c410-f3r** [comment]
Thank you guys!

**Comment by c410-f3r** [comment]
> Socket io Timed out: 60 secs waiting for response. API_KEY=1001, CorrelationId=5

Error looks spurious

**Comment by morenol** [comment]
bors retry

**Comment by c410-f3r** [comment]
Thank you @morenol !

**Pull request closed without merging**